odl/trafos/fourier.py

```diff
@@ -0,0 +1,1024 @@
+    planning_timelimit : `float`, optional
+        Limit planning time to roughly this amount of seconds.
+        Default: `None` (no limit)
+    threads : `int`, optional
+        Number of threads to use. Default: 1
+    normalise_idft : `bool`, optional
+        If `True`, the backward transform is normalized by
+        ``1 / N``, where ``N`` is the total number of points in
+        ``array_in[axes]``. This ensures that the IDFT is the true
+        inverse of the forward DFT.
+        Default: `False`
+    import_wisdom : `str`, optional
+        File name to load FFTW wisdom from. If the file does not exist,
+        it is ignored.
+    export_wisdom : `str`, optional
+        File name to append the accumulated FFTW wisdom to
+
+    Returns
+    -------
+    fftw_plan : ``pyfftw.FFTW``
+        The plan object created from the input arguments. It can be
+        reused for transforms of the same size with the same data types.
+        Note that reuse only gives a speedup if the initial plan
+        used a planner flag other than ``'estimate'``.
+        If ``fftw_plan`` was specified, the returned object is a
+        reference to it.
+
+    Notes
+    -----
+    * The planning and direction flags can also be specified as
+      capitalized and prepended by ``'FFTW_'``, i.e. in the original
+      FFTW form.
+    * For a ``halfcomplex`` forward transform, the arrays must fulfill
+      ``array_out[axes].shape[-1] == array_in[axes].shape[-1] // 2 + 1``,
+      and vice versa for backward transforms.
+    * All planning schemes except ``'estimate'`` require an internal copy
+      of the input array but are often several times faster after the
+      first call (measuring results are cached). Typically,
+      'measure' is a good compromise. If you cannot afford the copy,
+      use 'estimate'.
+    * The input can be destroyed if a planner different from
+      ``'estimate'`` is used or a complex-to-real backward transform is
+      computed.
+    """
+    import pickle
+
+    if not array_in.flags.aligned:
+        raise ValueError('Input array not aligned.')
+
+    if not array_out.flags.aligned:
+        raise ValueError('Output array not aligned.')
+
+    # We can use _fftw_to_local here since it strigifies and converts to
+    # lowercase
+    if axes is None:
+        axes = tuple(range(array_in.ndim))
+    else:
+        axes = tuple(axes)
+
+    direction = _pyfftw_to_local(direction)
+    fftw_plan = kwargs.pop('fftw_plan', None)
+    planning_effort = _pyfftw_to_local(kwargs.pop('planning_effort',
+                                                  'estimate'))
+    planning_timelimit = kwargs.pop('planning_timelimit', None)
+    threads = kwargs.pop('threads', 1)
+    normalise_idft = kwargs.pop('normalise_idft', False)
+    wimport = kwargs.pop('import_wisdom', '')
+    wexport = kwargs.pop('export_wisdom', '')
+
+    # Cast input to complex if necessary
+    array_in_copied = False
```

Comment:
This should be a flag that users can provide as `True` in case they have done this externally.